### PR TITLE
EX-2549 Dealids custom param

### PR DESCRIPTION
**Pull request by egsgordeev**
## Type of change
- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Pass the segments parameter value as imp.ext.dealids array

### CODE REVIEW

File: modules/sovrnBidAdapter.js [hunk 1]
```diff
@@ -88,6 +88,14 @@ export const spec = {
         };
       }
 
+      let segmentsString = utils.getBidIdParameter('segments', bid.params)
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: should be const

File: modules/sovrnBidAdapter.js [hunk 2]
```diff
@@ -88,6 +88,14 @@ export const spec = {
...
+
+      if (segmentsStr) {
+        let dealids = segmentsStr.split(',');
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: should also be a const

**Review by jrosendahl — APPROVED** [review]
Two nitpicks

File: modules/sovrnBidAdapter.js (hunk 1 above)
Comment (egsgordeev) [reply modules/sovrnBidAdapter.js]: Done!

File: modules/sovrnBidAdapter.js (hunk 2 above)
Comment (egsgordeev) [reply modules/sovrnBidAdapter.js]: Done!

**Comment by egsgordeev** [comment]
@jrosendahl 
The tests mostly stayed the same since yesterday, the only change is at the 202nd line: it now expects `ext` to be `undefined` 

Also the OpenRTB field is now called `deals` as we wait for such field in Blackbird as is.

File: modules/sovrnBidAdapter.js [hunk 3]
```diff
@@ -63,7 +64,17 @@ export const spec = {
           },
           tagid: String(utils.getBidIdParameter('tagid', bid.params)),
           bidfloor: utils.getBidIdParameter('bidfloor', bid.params)
-        });
+        }
+
+        sovrnImps.push(imp);
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: it does not actually matter but I would prefer to see the push after the variable is complete.

File: modules/sovrnBidAdapter.js [hunk 4]
```diff
@@ -53,7 +54,7 @@ export const spec = {
         bidSizes = ((utils.isArray(bidSizes) && utils.isArray(bidSizes[0])) ? bidSizes : [bidSizes])
         bidSizes = bidSizes.filter(size => utils.isArray(size))
         const processedSizes = bidSizes.map(size => ({w: parseInt(size[0], 10), h: parseInt(size[1], 10)}))
-        sovrnImps.push({
+        imp = {
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: This should be a const

File: modules/sovrnBidAdapter.js [hunk 5]
```diff
@@ -30,6 +30,7 @@ export const spec = {
       let eids;
       let tpid = []
       let criteoId;
+      let imp;
```
Comment (jrosendahl) [inline modules/sovrnBidAdapter.js]: Imp is not used outside of the _each scope so it should be defined there.

**Review by jrosendahl — APPROVED** [review]
More nits - nice catch on the deals / dealids

File: modules/sovrnBidAdapter.js (hunk 3 above)
Comment (egsgordeev) [reply modules/sovrnBidAdapter.js]: Moved to the right place, thank you!

File: modules/sovrnBidAdapter.js (hunk 4 above)
Comment (egsgordeev) [reply modules/sovrnBidAdapter.js]: Done!

File: modules/sovrnBidAdapter.js (hunk 5 above)
Comment (egsgordeev) [reply modules/sovrnBidAdapter.js]: Done!

**Review by sejanick — APPROVED** [review]
LGTM!

**Pull request merged**